### PR TITLE
Adding LightGlue

### PR DESCRIPTION
**Pull request by ducha-aiki**
#### Changes
Adds LightGlue.

#### Type of change

- [x] 📚  Documentation Update
- [x] 🧪 Tests Cases
- [x] 🔬 New feature (non-breaking change which adds functionality)

#### Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [] My changes generate no new warnings
- [ ] Did you update CHANGELOG in case of a major change?

### CODE REVIEW

File: kornia/feature/integrated.py [hunk 1]
```diff
@@ -393,3 +395,67 @@ def forward(self, data: Dict[str, Tensor]) -> Dict[str, Tensor]:
             'confidence': concatenate(out_confidence, dim=0).view(-1),
             'batch_indexes': concatenate(out_batch_indexes, dim=0).view(-1),
         }
+
+
+class LightGlueMatcher(GeometryAwareDescriptorMatcher):
+    """LightGlue-based matcher  in kornia API.
```
Comment (edgarriba) [inline kornia/feature/integrated.py:401]:
```suggestion
    """LightGlue-based matcher  in kornia API.
 
```

**Review by edgarriba — APPROVED** [review]
LGTM. Any demo or tutorial update would be nice for later. Is this the kinda last top model or still Loftr is the thing ?

**Comment by ducha-aiki** [comment]
Loftr is still the thing, however, on some images LG is better and it definitely much faster.

**Comment by edgarriba** [comment]
> Loftr is still the thing, however, on some images LG is better and it definitely much faster.

would be great to update the tutorials or image matching page to explain when every model we have is useful

**Comment by ducha-aiki** [comment]
>would be great to update the tutorials or image matching page to explain when every model we have is useful
and probably merge the most of image matching tutorials into single one, explaining the differences.

**Comment by ducha-aiki** [comment]
Closing in favour of https://github.com/kornia/kornia/pull/2449

**Pull request closed without merging**